Plugin/AdaptivePixelRatioPlugin.php

```diff
@@ -80,7 +80,8 @@ public function beforeToHtml(Image $subject)
         }
 
         if (version_compare($this->productMetadata->getVersion(), '2.4', '<')) {
```

Comment:
Can you implement something like this? We need to check if `custom_attributes` is set and also we can avoid the extra empty space if it is not set.
```php
$srcSet = implode(',', $srcSet);
if (version_compare($this->productMetadata->getVersion(), '2.4', '<')) {
    $attributes = ['srcset="' . $srcSet . '"'];
    if ($subject->getData('custom_attributes')) {
        $attributes[] = $subject->getData('custom_attributes');
    }
    $subject->setData('custom_attributes', implode(' ', $attributes));
} else {
    $customAttributes = $subject->getCustomAttributes() ?: [];
    $customAttributes['srcset'] = $srcSet;
    $subject->setData('custom_attributes', $customAttributes);
}
```

Reply:
This looks a bit cleaner, I agree! Will change that :)

Reply:
I have changed the logic a bit, on my environment it works perfectly now. What do you think?